src/opa.js

```diff
@@ -153,16 +189,24 @@ async function _load_policy(policy_wasm, memory) {
   return wasm;
 }
 
-// LoadedPolicy is a wrapper around a WebAssembly.Instance and WebAssembly.Memory
-// for a compiled Rego policy. There are helpers to run the wasm instance and
-// handle the output from the policy wasm.
+/**
+ * LoadedPolicy is a wrapper around a WebAssembly.Instance and WebAssembly.Memory
+ * for a compiled Rego policy. There are helpers to run the wasm instance and
+ * handle the output from the policy wasm.
+ */
 class LoadedPolicy {
+  /**
+   * Loads and initializes a compiled Rego policy.
+   * @param {WebAssembly.WebAssemblyInstantiatedSource} policy 
+   * @param {WebAssembly.Memory} memory 
+   */
   constructor(policy, memory) {
     this.mem = memory;
 
     // Depending on how the wasm was instantiated "policy" might be a
     // WebAssembly Instance or be a wrapper around the Module and
     // Instance. We only care about the Instance.
+    /** @type {OpaInstance} */
```

Comment:
Do we need this type information here? Playing around with the code I removed the line and my IDE didn't flag anything as suspicious and the e2e test still worked.

If there is value in it I'm all for keeping them, but any of these type comments in the middle of code blocks are going to be very high risk of getting out of sync.

Reply:
It causes the output type of `new LoadedPolicy().wasmInstance` to be a `OpaInstance`. Otherwise TypeScript can't really parse that line. Removed it if only that _here be dragons_ if an external consumer tries to mess with the underlying wasm instance.